core/src/providers/openai.rs

```diff
@@ -1440,24 +1440,32 @@ impl LLM for OpenAILLM {
     }
 
     fn context_size(&self) -> usize {
-        if self.id.starts_with("gpt-3.5-turbo") {
-            return 4096;
-        }
-        if self.id.starts_with("gpt-4-32k") {
-            return 32768;
-        }
-        if self.id.starts_with("gpt-4-1106-preview") {
-            return 128000;
+        // Reference: https://platform.openai.com/docs/models
+
+        // gpt-3.5-*
+        if self.id.starts_with("gpt-3.5") {
```

Comment:
Duplicated code, we can call this function from the Azure code, right?

Reply:
Sure I'll create a function :p

Reply:
(done)